### PR TITLE
provide more useful ambiguous-identifier error output

### PR DESCRIPTION
**Pull request by jrfondren**
The following code fails to compile with the following error:
```nim
import strutils

void(join)
```

### Current Output
```
ambjoin.nim(3, 5) Error: ambiguous identifier: 'join' --use strutils.join or strutils.join
```

### Expected Output
Something more like
```
ambjoin.nim(3, 5) Error: ambiguous identifier: 'join' -- use one of the following:
  strutils.join: proc (a: openarray[string], sep: string): string{.noSideEffect, gcsafe, locks: 0.}
  strutils.join: proc (a: openarray[T: not string], sep: string): string{.noSideEffect, gcsafe.}
```

### Possible Solution
I looked a little bit into compiler/lookups.nim:errorUseQualifier (which produces the current output) vs. compiler/semexprs.nim:semIndirectOp (which produces the more-desirable output), and ... turned this into a PR instead of an issue. The provided change results in the 'expected output'.

### Additional Information
The current output is seen with devel and with 0.19.4

### CODE REVIEW

**Comment by kaushalmodi** [comment]
I like this idea, except that the CI is failing. 

A minor suggestion, can you add a little hanging indentation (2 spaces) to lines 2,3 so that it's more evident that that's a continuation of the first line?

E.g.

```
ambjoin.nim(3, 5) Error: ambiguous identifier: 'join' --use one of the following: 
  strutils.join: proc (a: openarray[string], sep: string): string{.noSideEffect, gcsafe, locks: 0.} 
  strutils.join: proc (a: openarray[T: not string], sep: string): string{.noSideEffect, gcsafe.} 
```

**Comment by jrfondren** [comment]
@kaushalmodi aye, that looks nicer--added.  I don't think there's a lot of indent in compiler output, but it does occur in
```
proc `+`(x, y: float32): float32
  first type mismatch at position: 1
  required type: float32
  but expression 'void' is of type: type void
```
and `grep -r '\\n  ' compiler/` sees a few others, all two spaces like this.

File: compiler/lookups.nim [hunk 1]
```diff
@@ -255,9 +255,10 @@ proc errorUseQualifier*(c: PContext; info: TLineInfo; s: PSym) =
   var candidate = initIdentIter(ti, c.importTable.symbols, s.name)
   var i = 0
   while candidate != nil:
-    if i == 0: err.add " --use "
-    else: err.add " or "
-    err.add candidate.owner.name.s & "." & candidate.name.s
+    if i == 0: err.add " --use one of the following:\n"
```
Comment (bluenote10) [inline compiler/lookups.nim]: While you are at it, could you add space after the `--`? It's meant as a long-dash, which typically has symmetric whitespace. My brain was parsing `--use` as a command line switch all the time ;).

Comment (jrfondren) [reply compiler/lookups.nim]: @bluenote10 done, thanks. I've updated the 'expected output' above.

**Comment by jrfondren** [comment]
latest build failures look to be unrelated. The failing test was matching the text of the error that this PR changes.

**Pull request merged**